lib/wings/model_transformer.rb

```diff
@@ -196,11 +196,37 @@ def internal_resource
 
     class AttributeTransformer
       def self.run(obj, keys)
-        keys.each_with_object({}) do |attr_name, mem|
-          next unless obj.respond_to? attr_name
+        # TODO: There is an open question about whether we want to treat all these relationships the same.  See Issue #3904.
+        attrs = keys.select { |k| k.to_s.end_with? '_ids' }.each_with_object({}) do |attr_name, mem|
+          mem[attr_name.to_sym] =
+            TransformerValueMapper.for(obj.try(attr_name)).result ||
+            TransformerValueMapper.for(attr_ids_for_name_minus_ids(obj, attr_name)).result ||
+            TransformerValueMapper.for(attr_ids_for_name_minus_ids_pluralized(obj, attr_name)).result || []
+        end
+        keys.each_with_object(attrs) do |attr_name, mem|
+          next unless obj.respond_to?(attr_name) && !mem.key?(attr_name.to_sym)
           mem[attr_name.to_sym] = TransformerValueMapper.for(obj.public_send(attr_name)).result
         end
       end
+
+      def self.attr_ids_for_name_minus_ids(obj, attr_name)
+        real_attr_name = attr_name_minus_ids(attr_name)
+        attr_value = obj.try(real_attr_name)
+        return [attr_value.id] if attr_value.respond_to? :id
+        return attr_value.map(&:id) if attr_value.respond_to? :each
+      end
+
+      def self.attr_ids_for_name_minus_ids_pluralized(obj, attr_name)
+        real_attr_name = attr_name_minus_ids(attr_name) + 's'
+        attr_value = obj.try(real_attr_name)
+        return [attr_value.id] if attr_value.respond_to? :id
+        return attr_value.map(&:id) if attr_value.respond_to? :each
+      end
+
+      def self.attr_name_minus_ids(attr_name)
+        return unless attr_name.ends_with? '_ids'
```

Comment:
why do we want to return `nil`? if we can tolerate returning the string in the case the string doesn't end with `"_ids"`, this is simply `attr_name.chomp('_ids')`.